KubeVirt Use Case Documentation

davidvossel opened the pull request:
Here's my shot at trying to define the use cases and scope we're targeting for KubeVirt.

I found it difficult to describe what we're doing without comparing the projects to other things within the ecosystem people might be familiar with. My hope was by doing this comparison people would be able to more firmly grasp the scope of the project. Otherwise, by just listing use cases I though it was difficult to differentiate what we're doing from everything else.

hopefully this is all on target

stu-gott commented on docs/use-cases.md:
```diff
@@ -0,0 +1,58 @@
+
+## OpenStack
+
+OpenStack is an open source IaaS platform built of various components to provide a wide range of features such as compute (vm scheduling), networking, block storage, object storage, ect. 
```
Minor nit: Capitalization of VM is inconsistent throughout.

stu-gott commented on docs/use-cases.md:
```diff
@@ -0,0 +1,58 @@
+
+**Kubernetes Trusted Workloads** - The ability to run virtualized workloads in a Kubernetes cluster that are unsafe to run without the security guarantees a hypervisor provides.
+
+**Combining Container and Virtualized Workloads** - The ability to schedule both container and virtualized workloads on the same Kubernetes cluster. 
```
Minor nit: trailing whitespace in a few places.

rmohr commented on docs/use-cases.md:
```diff
@@ -0,0 +1,58 @@
+
+Hopefully it is obvious that KubeVirt is not a service. Kubevirt is an opensource project that focuses solely on scheduling VMs onto a cluster of machines and is not tightly coupled to any network or storage backend.  When compared to EC2 and GCE, we view KubeVirt’s scope to be that of the scheduling component provided by these services.
+
+We aim to provide the kind of feature set and flexibility that would allow one these services to replace their VM scheduling component with KubeVirt.
```
s/one//

rmohr commented on docs/use-cases.md:
```diff
@@ -0,0 +1,58 @@
+
+KubeVirt aims to provide a featureset capable of providing the strong consistency guarantees required for pet VMs as well as the scale out functionality desired for cloud IaaS. Compared to oVirt, KubeVirt could replace the scheduling component of oVirt and consume the network and storage resources oVirt already provides.
+
+## Kubernetes
```
Maybe we should put kubernetes on top of the comparisons?

davidvossel replied:
agreed

mykaul commented on docs/use-cases.md:
```diff
@@ -0,0 +1,59 @@
+
+KubeVirt has made the choice to focus on KVM managed by Libvirt. Initially this may sound like it is limiting KubeVirt’s functionality, but in practice we see KVM and Libvirt emerging as the leading standard in open source virtualization. By focusing on this technology stack we aren’t limited by creating a one size fits all virtualization technology abstraction.  We can fully support a feature se ...
+
+## oVirt and RHEV
```
RHV, or Red Hat Virtualization

berrange replied:
Just remove the reference to RHEV entirely, and talk about oVirt exclusively.   RHEV is just a particular vendor's productization of oVirt so its redundant to list it. That avoids us chasing whatever preferred product naming is this week.

davidvossel replied:
agreed

berrange commented on docs/use-cases.md:
```diff
@@ -0,0 +1,59 @@
+# Overview
+
+KubeVirt is a tool for scheduling and managing the lifecycle of Virtual Machines (VMs) across a distributed set of nodes.  In a broad sense KubeVirt is targeting all use-cases involving VM placement and management in the datacenter.
+
```
Can you line wrap the doc at max 80 chars

davidvossel replied:
yup

berrange commented on docs/use-cases.md:
```diff
@@ -0,0 +1,59 @@
+
+Nova is the open source component of OpenStack responsible for VM scheduling. Nova provides an abstraction layer that allows multiple virtualization technologies to be utilized. This means Nova is maintaining drivers for technologies such as Xen, VMware, KVM, Hyper-V, and LXC.
+
+KubeVirt has made the choice to focus on KVM managed by Libvirt. Initially this may sound like it is limiting KubeVirt’s functionality, but in practice we see KVM and Libvirt emerging as the leading standard in open source virtualization. By focusing on this technology stack we aren’t limited by creating a one size fits all virtualization technology abstraction.  We can fully support a feature se ...
```
s/we see KVM and Libvirt emerging/KVM and Libvirt have emerged/

KVM + libvirt has been an order of magnitude more popular than any other hypervisor in openstack user surveys for years now.

berrange commented on docs/use-cases.md:
```diff
@@ -0,0 +1,59 @@
+
+Cloud infrastructure provided by OpenStack or public clouds like EC2 and GCE make the assumption that applications running on their platforms have resiliency built in.  oVirt is for the use cases where applications require strict HA guarantees at the infrastructure level. With oVirt, data integrity is enforced by node level fencing.
+
+This comes down to the “pets vs. cattle” debate.  oVirt is capable of providing the kinds of guarantees users need to run pet VMs while other projects are focused on cattle.
```
I would prefer to avoid mention "pets" vs "cattle", as I find it has been really counterproductive in discussions - both oVirt and OpenStack are able to host either kind of workload.

davidvossel replied:
i'm just going to remove this sentence entirely.

davidvossel commented:
Thanks for the feedback. I updated the doc.

davidvossel commented:
anyone else want to weigh in on this? i'd like to get this pr closed.  If there's no additional feedback by the end of the week I plan on merging this.

berrange commented:
The content looks fine, but I think I'd like to have this in the website repo instead, since it is a key user-facing information to be presented.

davidvossel commented:
I've addressed the latest round of feedback and moved this pull request to the website.

moved: https://github.com/kubevirt/kubevirt.github.io/pull/6
